docs/06-using-cairo.md

```diff
@@ -0,0 +1,160 @@
+
+    print("This will print a table address:", require("lgi").cairo)
+    print("This will print an error:", require("lgi").Cairo)
+
+It is recommanded to avoid using `require` everytime you use a function and
+instead include the libraries at the top of your `rc.lua` or lua module
+
+    local cairo = require("lgi").cairo
+
+## The Cairo API
+
+Cairo is a 2D graphic library used by Awesome, Gnome and XFCE. It allows to
+paint paths on a `surface`. Awesome use it internally and being able to call
+is directly should be considered a power user feature.
+
+Understanding these few important concepts is necessary to be able to use Cairo:
+
+**Surface**:
+
+A surface is the area where the painting will be done. There is multiple types
+of surfaces including:
+
+ * Color pixmap with transparency (`ARGB32`) or without (`RGB24`)
+ * Monochrome pixmap surfaces with transparency (`A8`) or without (`A1`)
+ * SVG vectorial surfaces
+ * Native (XCB) surfaces
+ * Framebuffers and other less interesting ones from an AwesomeWM point of view.
+
+For more details, read:
+
+* [Surfaces](https://cairographics.org/manual/cairo-Image-Surfaces.html).
+
+**Sources**:
+
+Sources are elements like colors, patterns or gradient. See `gears.color` for
+common sources.
+
+For more details, read:
+
+* [Pattern](http://cairographics.org/manual/cairo-cairo-pattern-t.html)
+
+**Context and paths**:
+
+A context is the proxy between the program and the surface. The context hold a
+set of paths. Paths are something like a line, circle or rectangle. They may or
+may not be closed (a shape). All operations on the surfaces has to be told to
+the context. Until the context state is flushed (see next section), those paths
+add up, until that, nothing will be written to the surface. For example:
+
+    cr:rectangle(0 , 0 , 10, 10)
+    cr:rectangle(10, 10, 10, 10)
+
+wont do anything until and operation is applied on the context.
+
+For more details, read:
+
+* [Path](http://cairographics.org/manual/cairo-Paths.html)
+* [Context](http://cairographics.org/manual/cairo-cairo-t.html)
+* [Transformation](http://cairographics.org/manual/cairo-Transformations.html)
+
+A context also hold a transformation matrix (see `gears.matrix`) used when
+applying an operation.
+
+**Operations**:
+
+Multiple operations can be done with the paths. The most common are:
+
+ * *fill*: Fill the path with the current source
+ * *stroke*: Paint the path outline with the current source
+ * *mask*: Use the current source as an alpha mask while painting with the
+   current operator.
+ * *clip*: Crop the surfaces workarea so nothing outside of the clip will be
+   affected by the next opetations.
+
+**Operators**:
+
+[Operators](http://cairographics.org/operators/) are modifiers used when
+applying operations.
+
+### Cairo in Awesome
+
+Each Awsome `wibox`es, `awful.wibar`s, `gears.wallpaper`s and `awful.titlebar`s
+are in fact glorified Cairo surfaces. They can be used as such using an hidden
```

Comment:
Technically they are not, they contain one... but whatever...

an hidden -> a hidden
(Is it really hidden? The API docs contain it)